doc/exceptions.md

```diff
@@ -0,0 +1,365 @@
+| <tt>end</tt> statement.     | Ends the handler.  `                                                                     |
+
+#### Begin Clause
+
+The begin clause begins the exception handler:
+
+- May start with a `begin` statement;
+  see also [Begin-Less Exception Handlers](#label-Begin-Less+Exception+Handlers).
+- Contains code whose raised exception (if any) is covered
+  by the handler.
+- Ends with the first following `rescue` statement.
+
+#### Rescue Clauses
+
+A rescue clause:
+
+- Starts with a `rescue` statement.
+- Contains code that is to be executed for certain raised exceptions.
+- Ends with the first following `rescue`,
+  `else`, `ensure`, or `end` statement.
+
+A `rescue` statement may include one or more classes
+that are to be rescued;
+if none is given, StandardError is assumed.
+
+The rescue clause rescues both the specified class
+(or StandardError if none given) or any of its subclasses;
+(see [Built-In Exception Classes](rdoc-ref:Exception@Built-In+Exception+Classes)
+for the hierarchy of Ruby built-in exception classes):
+
+
+```
+begin
+  1 / 0 # Raises ZeroDivisionError, a subclass of StandardError.
+rescue
+  puts "Rescued #{$!.class}"
+end
+```
+
+Output:
+
+```
+Rescued ZeroDivisionError
+```
+
+If the `rescue` statement specifies an exception class,
+only that class (or one of its subclasses) is rescued;
+this example exits with a ZeroDivisionError,
+which was not rescued because it is not ArgumentError or one of its subclasses:
+
+```
+begin
+  1 / 0
+rescue ArgumentError
+  puts "Rescued #{$!.class}"
+end
+```
+
+A `rescue` statement may specify multiple classes,
+which means that its code rescues an exception
+of any of the given classes (or their subclasses):
+
+```
+begin
+  1 / 0
+rescue FloatDomainError, ZeroDivisionError
+  puts "Rescued #{$!.class}"
+end
+```
+
+An exception handler may contain multiple rescue clauses;
+in that case, the first clause that rescues the exception does so,
+and those before and after are ignored:
+
+```
+begin
+  Dir.open('nosuch')
+rescue Errno::ENOTDIR
+  puts "Rescued #{$!.class}"
+rescue Errno::ENOENT
+  puts "Rescued #{$!.class}"
+end
+```
+
+Output:
+
+```
+Rescued Errno::ENOENT
+```
+
+A `rescue` statement may specify a variable
+whose value becomes the rescued exception
+(an instance of Exception or one of its subclasses:
+
+```
+begin
+  1 / 0
+rescue => x
+  puts x.class
+  puts x.message
+end
+```
+
+Output:
+
+```
+ZeroDivisionError
+divided by 0
+```
+
+In the rescue clause, these global variables are defined:
+
+- `$!`": the current exception instance.
+- `$@`: its backtrace.
+
+#### Else Clause
+
+The `else` clause:
+
+- Starts with an `else` statement.
+- Contains code that is to be executed if no exception is raised in the begin clause.
+- Ends with the first following `ensure` or `end` statement.
+
+```
+begin
+  puts 'Begin.'
+rescue NoSuch
```

Comment:
I think it's confusing and not relevant to use a nonexistent exception here. We could just use an empty rescue clause I think

```suggestion
rescue
```

Reply:
Done.